### PR TITLE
Rating improvement hard fork stuck shard

### PR DESCRIPTION
**Pull request by mariusmihaic**
- Added `MaxConsecutiveRoundsOfRatingDecrease` , which represents the max number of consecutive rounds in which a validator's rating can be decrease in case of not processing/signing a block. If, for instance, a shard gets stuck for more rounds than this value, then we should stop decreasing validator's rating so they won't get jailed.
- Added `StopDecreasingValidatorRatingWhenStuckEnableEpoch` which represents the epoch when we should stop decreasing validator's rating if, for instance, a shard gets stuck
- Added simple logic to check if enable epoch flag is set AND missed rounds > `MaxConsecutiveRoundsOfRatingDecrease` => do not decrease validator rating
- Added unit test

### CODE REVIEW

File: process/errors.go [hunk 1]
```diff
@@ -569,6 +569,9 @@ var ErrNilEconomicsData = errors.New("nil economics data")
 // ErrZeroMaxComputableRounds signals that a value of zero was provided on the maxComputableRounds
 var ErrZeroMaxComputableRounds = errors.New("max computable rounds is zero")
 
+// ErrZeroMaxConsecutiveRoundsOfRatingDecrease signals that a value of zero was provided on the MaxConsecutiveRoundsOfRatingDecrease
+var ErrZeroMaxConsecutiveRoundsOfRatingDecrease = errors.New("max consecutive number of rounds in which we can decrease a validator rating is zero")
```
Comment (raduchis) [inline process/errors.go:573]: max consecutive number of rounds**,** in which we can decrease a validator rating**,** is zero

**Review by raduchis — CHANGES_REQUESTED** [review]

File: cmd/node/config/config.toml [hunk 2]
```diff
@@ -23,9 +23,10 @@
    # GenesisMaxNumberOfShards represents the maximum number of shards to be created at genesis (excluding metaChain shard)
    GenesisMaxNumberOfShards  = 3
 
-   # MaxConsecutiveRoundsOfRatingDecrease represents the max number of consecutive rounds in which a validator's rating
-   # can be decrease in case of not processing/signing a block. If, for instance, a shard gets stuck for more rounds
-   # than this value, then we should stop decreasing validator's rating so they won't get jailed.
+   # MaxConsecutiveRoundsOfRatingDecrease represents the max number of consecutive rounds in which a block is not proposed
+   # on a shard and and the validators' rating could be decreased. If, for instance, a shard gets stuck for more rounds
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/node/config/config.toml:27]: Good catch, fixed it

File: cmd/node/config/config.toml [hunk 3]
```diff
@@ -23,9 +23,10 @@
...
+   # than this value, then we should stop decreasing validator's rating so they won't get jailed => then the validators
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/node/config/config.toml:28]: Renamed it !

File: process/errors.go (hunk 1 above)
Comment (mariusmihaic) [reply process/errors.go:573]: Added **,**

Comment (raduchis) [reply process/errors.go]: ok!

File: cmd/node/config/config.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply cmd/node/config/config.toml]: ok

File: cmd/node/config/config.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply cmd/node/config/config.toml]: ok

**Review by raduchis — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by raduchis — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**